### ISSUE TITLE
[Strict] Diagnose `structValue == null`

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#30196]
Given a user-defined type that overloads operator `==` and `!=`

``` c#
struct S1
{
    public static bool operator ==(S1 self, S1 other) => true;
    public static bool operator !=(S1 self, S1 other) => false;
}
```

The Roslyn compiler does not diagnose the following code:

``` c#
    void M(S1 s)
    {
        if (s == null) { } // warning: always false
        if (s != null) { } // warning: always true
        if (null == s) { } // warning: always false
        if (null != s) { } // warning: always true
    }
```

In "strict" mode the compiler warns about these tests.

Please add this warning to a warning wave.

**Comment by Youssef1313** [comment]
I think I have seen an issue before about having all strict warnings in warning wave. In that case, this issue will be automatically fixed as part of this and there is no need to explicitly add a warning wave for this. Can someone confirm if that's correct?

**Comment by gafter** [comment]
Closing as a dup of https://github.com/dotnet/roslyn/issues/45744. This warning is included in C# 9 if you use /warn:5 (the first warning wave)
